src/fparser/common/tests/test_readfortran.py

```diff
@@ -858,7 +858,28 @@ def test_string_reader():
         assert unit_under_test.get_single_line(ignore_empty=True) == expected
 
 
-##############################################################################
+@pytest.mark.parametrize("reader_cls", [FortranStringReader, FortranFileReader])
+def test_reader_ignore_encoding(reader_cls, tmp_path):
+    """
+    Tests that the Fortran{String,File}Reader can be configured to take notice of
+    Python-style encoding information.
+    """
+    source = "! -*- f77 -*-\n" + FULL_FREE_SOURCE
+    if reader_cls is FortranFileReader:
+        sfile = tmp_path / "my_test.f90"
+        sfile.write_text(source)
+        # File location with full path.
+        rinput = str(sfile.absolute())
+    else:
+        rinput = source
+    reader = reader_cls(rinput)
+    # By default the encoding information is ignored so the format should be
+    # free format, not strict.
+    assert reader.format == FortranFormat(True, False)
+    # Now test when the reader takes notice of the encoding information.
+    reader2 = reader_cls(rinput, ignore_encoding=False)
```

Comment:
Maybe I am overdoing it, but just in case maybe add a test for `ignore_ending=True` here.

Reply:
You can never have too many tests (within reason).

Reply:
I hope that was within reason .. thanks!